### DISCUSSION TITLE
Add new command to jump to chunks instead of just words

### DISCUSSION THREAD

**Discussion by qwerty01** [helix-editor/helix#10016] [Ideas]
For context, this feature request is related to #510 and the implementation in #8875.

Currently the way the amp-like jump command works is by filtering by words. This is the best choice for the default command, however, I would like to propose a second command which works on what I would like to call "chunks". There are times where instead of jumping and selecting a whole word, selecting part of a word is desired. An example would be changing `OnceCell` to `OnceLock`, instead of jumping to the beginning of the word and selecting the whole thing, jumping to the second part of the word, `Cell` would allow a replacement to be done in 1 keystroke.

Current implementation:
![image](https://github.com/helix-editor/helix/assets/1034539/607904a7-ab29-4e38-8647-53a20070985e)

New command:
![image](https://github.com/helix-editor/helix/assets/1034539/03d2a05e-e0be-4ea2-a0a5-a5f7f214cb6c)

it would also add a new jump command:
![image](https://github.com/helix-editor/helix/assets/1034539/5cc11422-e9d7-483a-8b3f-49b88e772f31)

I've got a proof of concept [here](https://github.com/qwerty01/helix/tree/amp-chunks) for how it would work, I'll rebase it to 24.03 and create a PR once that version is released.

Some open questions to answer:
- Currently I have it set up so that underscores are included in the selection, that way you can delete a full part of a word in one stroke; replacement would require adding `_` back in, so I'm not sure if this is the right approach
- Would it be worth adding in some movement keys to also move by chunks? That would probably be best as a separate PR, but since the functionality would already be there, it might be worth considering
- Currently the way reverse searching is implemented is by reversing the iterator of characters. This works for all the other jump options (words, etc) since they're reversible, but *doesn't* work for chunks, since one chunk boundary is from a lowercase letter to an uppercase letter and not vice-versa, I added in a hack to swap the characters it's checking when it's searching for reverse chunks, but it might be worth finding a better solution.

**Comment by the-mikedavis** [comment]
IMO this is not a large enough change to warrant adding a new command. The jump mode we have from #8875 is not meant to be so precise that you can always jump to the exact position you want to get to. It's meant to be combined with other motions like t, f, w, etc. or searching `/`.

Maybe this behavior could be part of a larger change in the way we treat underscores (like for https://github.com/helix-editor/helix/issues/2429 / https://github.com/helix-editor/helix/issues/5188), so that customizing a word boundary could also change the behavior of `goto_word` / `extend_to_word`. But on its own I'm not interested in a new command.
